File: C++/프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.cpp
// #include <iostream>
#include <vector>

using namespace std;

int solution(vector<int> queue1, vector<int> queue2) {
    int answer = 0, i = 0;
    long queue_sum = 0, mid_sum = 0;
    int left = 0, right = queue1.size() - 1;
    vector<int> queue(queue1.size() + queue2.size());
    for(auto& node : queue1) {
        queue[i++] = node;
        queue_sum += node;
        mid_sum += node;
    }
    for(auto& node : queue2) {
        queue[i++] = node;
        mid_sum += node;
    }
    mid_sum /= 2;
    
    while(left <= right) {
        // cout << "left: " << left << ", right: " << right << endl;
        // cout << "Current Queue Sum: " << queue_sum << endl;
        if(queue_sum < mid_sum && right + 1 < queue.size()) {
            queue_sum += queue[++right];
        } else if (queue_sum > mid_sum) {
            queue_sum -= queue[left++];
        } else {
            break;
        }
        answer++;
    }
    
    return (mid_sum == queue_sum) ? answer : -1;
}

File: Python3/프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.py
def solution(queue1, queue2):
    answer = 0
    queue, queue_sum = queue1 + queue2, sum(queue1)
    mid_sum = sum(queue) // 2
    # print("Mid Sum:", mid_sum)
    left, right = 0, len(queue1) - 1
    
    while left <= right:
        # print("left:", left, ", right:", right)
        # print("Current Sum:", queue_sum)
        # print("Current Move:", answer)
        # print()

        if queue_sum < mid_sum and right + 1 < len(queue):
            right += 1
            queue_sum += queue[right]
        elif queue_sum > mid_sum:
            queue_sum -= queue[left]
            left += 1
        else:
            break
        answer += 1
    
    return answer if mid_sum == queue_sum else -1